### PR TITLE
Fix bug that would cause pdl encoder to import types conflicting with inline types.

### PR DESCRIPTION
**Pull request by karthikbalasub**
Also improves how imports for includes are handled.

### CODE REVIEW

File: generator-test/src/test/pegasus/com/linkedin/pegasus/generator/test/idl/enums/EscapedSymbols.pdl [hunk 1]
```diff
@@ -5,7 +5,6 @@ namespace com.linkedin.pegasus.generator.test.idl.enums
  */
 enum EscapedSymbols {
   ENUM,
-  `enum`,
```
Comment (nickibi) [inline generator-test/src/test/pegasus/com/linkedin/pegasus/generator/test/idl/enums/EscapedSymbols.pdl]: why do we delete this one?

Comment (karthikbalasub) [reply generator-test/src/test/pegasus/com/linkedin/pegasus/generator/test/idl/enums/EscapedSymbols.pdl]: This was an issue introduced in previous commit. enum cannot be used as it is reserved keyword in java. The template generation step fails for this.

Comment (evanw555) [reply generator-test/src/test/pegasus/com/linkedin/pegasus/generator/test/idl/enums/EscapedSymbols.pdl]: Oops

File: data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java [hunk 2]
```diff
@@ -782,7 +782,7 @@ private void gatherTypes(DataSchema schema, boolean isDeclaredInline, Set<Name>
         }
         for (NamedDataSchema include : recordSchema.getInclude())
         {
-          gatherTypes(include, true, encounteredTypes, inlinedTypes);
+          gatherTypes(include, recordSchema.isIncludeDeclaredInline(include), encounteredTypes, inlinedTypes);
```
Comment (evanw555) [inline data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java]: Nice catch.

File: data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java [hunk 3]
```diff
@@ -756,7 +756,7 @@ private void writeDocAndProperties(RecordDataSchema.Field field) throws IOExcept
    * @param inlinedTypes cumulative set of all inlined types in this schema (and its descendents).
```
Comment (evanw555) [inline data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java]: We should clarify that this set is a collection of simple names, not full names.

File: generator-test/src/test/pegasus/com/linkedin/pegasus/generator/test/idl/imports/InlineTypeConflict.pdl [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  /**
+   * Inline type can also conflict with types used for other fields.
+   */
+  inlineType2: record Note {
+    a: int
+  }
+
+  /**
+   * This field reference a type (Note) that conflicts with Note defined inline, and thus must use fully qualified name.
```
Comment (evanw555) [inline generator-test/src/test/pegasus/com/linkedin/pegasus/generator/test/idl/imports/InlineTypeConflict.pdl]: nit: `reference` -> `references`

**Pull request merged**